mean.hhs

```diff
@@ -0,0 +1,104 @@
+/**
+ * @author Jianan Lin (林家南)
+ * @param input - the list, matrix, tensor or structure that you want to get the mean(average) of
+ * @returns - a number representing the mean of all the elements of a list, matrix, tensor or other structure
+ * 
+ */
+
+
+*import math: is_number
+*import math: flatten
+*import math: ndim
+*import math: deep_copy
+
+
+function mean(input) {
+
+    // if there are no arguments, there is an error
+    if (arguments.length === 0) {
+        throw new Error('No argument given');
+    }
+
+    // if the first argument is a number, then we should get the mean of a list of numbers
+    if (is_number(arguments[0])) {
+
+        // determine whether there is invalid element
+        for (let i = 0; i < arguments.length; i++) {
+            if (!is_number(arguments[i])) {
+                throw new Error('First argument suggests a list, but there is a non number element in arguments.');
+            }
+        }
+
+        // sum_array is the sum, n is the number, result is sum / n
+        let sum_array = 0, n = 0;
+        for (let i = 0; i < arguments.length; i++) {
+            sum_array += arguments[i];
+            n++;
+        }
+
+        let result = sum_array / n;
+        return result;
+    }
+
+    //now we're dealing with non-numbers and matrix-like structures:
+    //declare raw_in, set it to a input clone if its a Mat otherwise input itself
+    let in_type = (input instanceof Mat);
+    let raw_in = (in_type) ? input.clone() : deep_copy(input);
+
```

Comment:
i would move `let dims = ndim(raw_in)` here, so its accessed to all block statements. otherwise, its only accessible within that 1 if statement (which we need to change the structure of)